src/router/weather.js

```diff
@@ -1,181 +1,45 @@
 'use strict';
 
 const express = require('express');
-const fetch = require('node-fetch');
 
 const router = express.Router();
 
-const { logger } = require('../utils');
+const { logger, UNITS, backends } = require('../utils');
+const OpenWeatherMap = require('../backend/OpenWeatherMap');
+const OpenMeteo = require('../backend/OpenMeteo');
 
-const OPEN_WEATHER_ROOT = 'https://api.openweathermap.org/data/2.5/onecall';
-const APP_ID = process.env.OWM_KEY;
+let BACKEND;
 
-const capitalize = (s) => s && s[0].toUpperCase() + s.slice(1);
-
-const getWeatherConditionOWM = (conditionCode) => {
-  if (conditionCode >= 200 && conditionCode < 600) {
-    return 'RAIN';
-  }
-  if (conditionCode >= 600 && conditionCode < 611) {
-    return 'SNOW';
-  }
-  if (conditionCode >= 611 && conditionCode < 700) {
-    return 'SLEET';
-  }
-  if (conditionCode >= 701 && conditionCode < 771) {
-    return 'FOG';
-  }
-  if (conditionCode >= 771 && conditionCode < 800) {
-    return 'WIND';
-  }
-  if (conditionCode === 800) {
-    return 'CLEAR';
-  }
-  if (conditionCode >= 801 && conditionCode < 803) {
-    return 'PARTLY_CLOUDY';
-  }
-  if (conditionCode >= 803 && conditionCode < 900) {
-    return 'CLOUDY';
-  }
-  return 'CLOUDY';
-};
-
-const tempUnit = (system) => {
-  switch (system) {
-    case 'IMPERIAL':
-      return 'F';
-    case 'METRIC':
-      return 'C';
+(() => {
+  switch (process.env.BACKEND) {
+    case backends.OWM:
+      BACKEND = new OpenWeatherMap();
+      break;
+    case backends.METEO:
+      BACKEND = new OpenMeteo();
+      break;
     default:
-      return 'K';
+      break;
   }
-};
-
-const visibilityUnit = (system) => (system === 'IMPERIAL' ? 'MI' : 'M');
+})();
```

Comment:
not sure what you're asking, this isn't in a function. this is a self-calling lambda...

Reply:
LOL, funny story. Since the time that I originally left this comment I learned about IIFEs (immediately invoked function expressions) and how they are used.  So disregard... :facepalm: